pkg/stis/calstis/cs2/cs2.c

```diff
@@ -23,7 +23,8 @@ int main (int argc, char **argv) {
 
 	c_irafinit (argc, argv);
 
-	if ((input = calloc (strlen(argv[1]) + 1, sizeof(char))) == NULL) {
+	/* ~1Kb padding per path, to allow for string modifications later */
+	if ((input = calloc(argc * (STIS_FNAME * 2), sizeof(char))) == NULL) {
```

Comment:
I would guess that it is an attempt to handle the worse case scenario that all (or significantly all) of the passed in arguments are filenames?
To me, this seems like a dangerous amount of stack allocation.
I have only briefly looked at the code but I'm not sure why this is the case at all.